content/posts/visualising-usage-using-batteries/index.md

```diff
@@ -0,0 +1,220 @@
+
+1. Outlining what we are going to plot
+2. Import necessary libraries
+3. Write a function to draw the battery
+    - This is the function that will be called to plot the battery chart
+4. Read the data and plot the chart accordingly
+    - We will demonstrate it with an example
+
+
+## <span style="text-decoration: underline">Plot Outline</span>
+
+What is our use case?
+
+- We are given a dataset where we have data of Liverpool's players and their minutes played in the last 2 seasons (for whichever club they played in that time period). We will use this data for our visualisation.
+- The final visualisation is the featured image of this blog post. We will navigate step-by-step to how we'll create the visualisation.
+
+## <span style="text-decoration: underline">Importing Libraries</span>
+
+The first and foremost part is to import the essential libraries so that we can leverage the functions within. In this case, we will import the following:-
+
+```python
+import pandas as pd
+import matplotlib.pyplot as plt
+from matplotlib.path import Path
+from matplotlib.patches import FancyBboxPatch, PathPatch, Wedge
+```
+
+The functions imported from `matplotlib.path` and `matplotlib.patches` will be used to draw lines, rectangles, boxes and so on to display the battery as it is.
+
+## <span style="text-decoration: underline">Drawing the Battery - A function</span>
+
+The next part is to define a function named `draw_battery()`,  which will be used to draw the battery. Later on, we will call this function by specifying certain parameters to build the figure as we require. The following below is the code to build the battery -
+
+```python
+def draw_battery(fig, ax, percentage=0, bat_ec="grey",
+                 tip_fc="none", tip_ec="grey", 
+                 bol_fc="#fdfdfd", bol_ec="grey", invert_perc=False):
+    '''
+    Parameters
+    ----------
+    fig : figure
+        The figure object for the plot
+    ax : axes
+        The axes/axis variable of the figure.
+    percentage : int, optional
+        This is the battery percentage - size of the fill. The default is 0.
+    bat_ec : str, optional
+        The edge color of the battery/cell. The default is "grey".
+    tip_fc : str, optional
+        The fill/face color of the tip of battery. The default is "none".
+    tip_ec : str, optional
+        The edge color of the tip of battery. The default is "grey".
+    bol_fc : str, optional
+        The fill/face color of the lighning bolt. The default is "#fdfdfd".
+    bol_ec : str, optional
+        The edge color of the lighning bolt. The default is "grey".
+    invert_perc : bool, optional
+        A flag to invert the percentage shown inside the battery. The default is False
+
+    Returns
+    -------
+    None.
+
+    '''
+    try:
+        fig.set_size_inches((15,15))
```

Comment:
Why are you doing this?

Reply:
I think I had left it during the early phase of writing the code. It isn't needed if we mention figure size properly later on while plotting. Setting the figure size to 15 x 15 seemed ideal for this case (with 25 players), so I had defined it early in the function.

It isn't needed in the function actually. If this line is omitted, we should probably modify line 183 from -

fig, ax = plt.subplots(5, 5, figsize=(5, 5))

to

fig, ax = plt.subplots(5, 5, figsize=(15, 15))

This is really upto the developer to decide what deems fit.